testcases/open_posix_testsuite/scripts/generate-makefiles.sh

```diff
@@ -190,6 +191,14 @@ run.sh:
 EOF
 	fi
 
+	link_libs=true
+	while getopts "c" optch $compiler_args; do
```

Comment:
getopts writes a line into stderr for each unknown option, which would spam the script output in a case that any other parameters are passed to this function.

Looking at the code, we do not pass anything but '-c' in the case of buildonly tests, so we may as well just do a simple string comparison, or if we want to be future proof I would prefer a simple for loop.

Reply:
I’d agree with you at the top-level, but other CFLAGS can sneak in on a directory by directory basis. I was trying to make this resistant against that scenario.

I could redirect stderr to /dev/null to mute the noise.

Reply:
Aren't all variables from files as CFLAGS added to Makefile as separate CFLAGS+=foo lines?

I checked the script and as far as I can tell, there is no way for compiler_args to contain anything else but '-c' in case of buildonly tests.

Did I missed something?

Reply:
You’re right. It’s early in PST..  I could set it to a variable and do a strict comparison. I was just trying to avoid making the logic brittle/too voodooey (it’s complicated enough as-is).